spring-integration-core/src/main/java/org/springframework/integration/channel/FluxSubscribableChannel.java

```diff
@@ -16,17 +16,18 @@
 
 package org.springframework.integration.channel;
 
-import org.reactivestreams.Publisher;
-
 import org.springframework.messaging.Message;
 
+import reactor.core.publisher.Flux;
+
 /**
  * @author Artem Bilan
+ * @author Gary Russell
  *
  * @since 5.0
  */
-public interface ReactiveSubscribableChannel {
+public interface FluxSubscribableChannel {
 
-	void subscribeTo(Publisher<Message<?>> publisher);
+	void subscribeTo(Flux<Message<?>> publisher);
```

Comment:
Yes using Publisher for input arguments and returning Flux/Mono (where one is already used internally) is a pattern we follow in Spring Framework APIs.